vlog4j-examples/src/main/java/org/semanticweb/vlog4j/examples/core/AddDataFromRdfFile.java

```diff
@@ -0,0 +1,170 @@
+		final Variable s = Expressions.makeVariable("s");
+		final Variable o = Expressions.makeVariable("o");
+
+		/*
+		 * We will write '<~/someName>' instead of <http://an.example/someName>.
+		 *
+		 * triplesIDB(?s, <~/bicycle>, ?o) :- triplesEDB(?s, <~/bicycle>, ?o) .
+		 */
+		final Atom bicycleIDB = Expressions.makeAtom(triplesIDB, s, bicyclePredicate, o);
+		final Atom bicycleEDB = Expressions.makeAtom(triplesEDB, s, bicyclePredicate, o);
+		final Rule rule1 = Expressions.makeRule(bicycleIDB, bicycleEDB);
+
+		/*
+		 * triplesIDB(?s, <~/wheel>, ?o) :- triplesEDB(?s, <~/wheel>, ?o) .
+		 */
+		final Atom wheelIDB = Expressions.makeAtom(triplesIDB, s, wheelPredicate, o);
+		final Atom wheelEDB = Expressions.makeAtom(triplesEDB, s, wheelPredicate, o);
+		final Rule rule2 = Expressions.makeRule(wheelIDB, wheelEDB);
+
+		/*
+		 * triplesIDB(?s, <~/hasPart>, ?o) :- triplesEDB(?s, <~/hasPart>, ?o) .
+		 */
+		final Atom hasPartIDB = Expressions.makeAtom(triplesIDB, s, hasPartPredicate, o);
+		final Atom hasPartEDB = Expressions.makeAtom(triplesEDB, s, hasPartPredicate, o);
+		final Rule rule3 = Expressions.makeRule(hasPartIDB, hasPartEDB);
+
+		/*
+		 * triplesIDB(?s, <~/isPartOf>, ?o) :- triplesEDB(?s, <~/isPartOf>, ?o) .
+		 */
+		final Atom isPartOfIDB = Expressions.makeAtom(triplesIDB, s, isPartOfPredicate, o);
+		final Atom isPartOfEDB = Expressions.makeAtom(triplesEDB, s, isPartOfPredicate, o);
+		final Rule rule4 = Expressions.makeRule(isPartOfIDB, isPartOfEDB);
+
+		/*
+		 * exists x. triplesIDB(?s, <~/hasPartIDB>, !x), triplesIDB(!x, <~/wheelIDB>,
+		 * <~/empty>) :- triplesIDB(?s, <~/bicycleIDB>, ?o) .
+		 */
+		final Atom existsHasPartIDB = Expressions.makeAtom(triplesIDB, s, hasPartPredicate, x);
+		final Atom existsWheelIDB = Expressions.makeAtom(triplesIDB, x, wheelPredicate, emptyObject);
+		final Rule rule5 = Expressions.makeRule(Expressions.makeConjunction(existsHasPartIDB, existsWheelIDB),
+				Expressions.makeConjunction(bicycleIDB));
+
+		/*
+		 * exists x. triplesIDB(?s, <~/isPartOfIDB>, !x) :- triplesIDB(?s, <~/wheelIDB>,
+		 * ?o) .
+		 */
+		final Atom existsIsPartOfIDB = Expressions.makeAtom(triplesIDB, s, isPartOfPredicate, x);
+		final Rule rule6 = Expressions.makeRule(Expressions.makeConjunction(existsIsPartOfIDB),
+				Expressions.makeConjunction(wheelIDB));
+
+		/*
+		 * triplesIDB(?s, <~/isPartOfIDB>, ?o) :- triplesIDB(?o, <~/hasPartIDB>, ?s) .
+		 */
+		final Atom hasPartIDBReversed = Expressions.makeAtom(triplesIDB, o, hasPartPredicate, s);
+		final Rule rule7 = Expressions.makeRule(isPartOfIDB, hasPartIDBReversed);
+
+		/*
+		 * triplesIDB(?s, <~/hasPartIDB>, ?o) :- triplesIDB(?o, <~/isPartOfIDB>, ?s) .
+		 */
+		final Atom isPartOfIDBReversed = Expressions.makeAtom(triplesIDB, o, isPartOfPredicate, s);
+		final Rule rule8 = Expressions.makeRule(hasPartIDB, isPartOfIDBReversed);
+
+		/* 2. Loading, reasoning, and querying. */
+		final Reasoner reasoner = Reasoner.getInstance();
```

Comment:
in examples project, it is a good idea to show how our system can be used most advantageously. Therefore, we should use try with resources whenever possible (perhaps except in one examples class, to display an alternative, and notify user to call reasoner.close()). 
Also, we can statically import Expressions.makeConstant, Expressions.makeVariable, Expressions.makeAtom etc, to obtain less lengthy code.